src/library.js

```diff
@@ -620,6 +617,53 @@ const createPilingJs = (rootElement, initOptions = {}) => {
     );
   };
 
+  const arrangePile = pile => {
+    const { orderer, arrangementType, arrangementObjective } = store.getState();
+
+    const getCellPosition = orderer(layout.numColumns);
+
+    let idx = null;
+    let x = null;
+    let y = null;
+    let i = null;
+    let j = null;
+    let u = null;
+    let v = null;
+
+    switch (arrangementType) {
+      case 'data':
+        break;
+      case 'index':
+        idx = arrangementObjective(pile);
+        [i, j] = getCellPosition(idx);
+        break;
+      case 'ij':
+        [i, j] = arrangementObjective(pile);
+        break;
+      case 'xy':
+        [x, y] = arrangementObjective(pile);
+        break;
+      case 'uv':
+        [u, v] = arrangementObjective(pile);
+        break;
+      // No automatic position
+      default:
+        break;
+    }
+
+    if (i !== null) {
+      [x, y] = layout.ijToXy(i, j, pile.graphics.width, pile.graphics.height);
+    } else if (u !== null) {
+      [x, y] = layout.uvToXy(u, v);
+    }
+
+    if (x !== null) {
+      store.dispatch(createAction.movePiles([{ id: pile.id, x, y }]));
```

Comment:
Doesn't this cause `movePiles` to be dispatched twice when this function is called by `positionPiles`?

Reply:
In `positionPiles`, if the return value `[x. y]` from `getPilePosition` is not null, it won't dispatch `movePiles` again. See line 703-705.